kernel/src/scan/mod.rs

```diff
@@ -407,7 +430,30 @@ impl Scan {
         &self,
         engine: &dyn Engine,
     ) -> DeltaResult<impl Iterator<Item = DeltaResult<ScanMetadata>>> {
-        self.scan_metadata_inner(engine, self.replay_for_scan_metadata(engine)?)
+        // Compute stats schema from the predicate's referenced columns if available.
+        // This ensures we only request stats for columns actually used in data skipping.
+        let stats_schema_opt = match &self.state_info.physical_predicate {
+            PhysicalPredicate::Some(_, referenced_schema) => build_stats_schema(referenced_schema),
+            _ => None,
+        };
+
+        let has_usable_stats_parsed = match stats_schema_opt {
+            Some(ref schema)
+                if self
+                    .snapshot
+                    .log_segment()
+                    .has_usable_stats_parsed(engine, schema) =>
+            {
+                true
+            }
+            _ => false,
+        };
+
+        self.scan_metadata_inner(
+            engine,
+            self.replay_for_scan_metadata(engine, stats_schema_opt, has_usable_stats_parsed)?,
```

Comment:
Should we push down the entire predicate here?  I think this would give the option to prune out unneeded parquet files by there footer stats if they had parsed stats?